File: 2021/12/03/parking_system.js
/**
 * https://leetcode.com/problems/design-parking-system/
 * @author Andrew Jarombek
 * @since 12/3/2021
 */

const ParkingSystem = function(big, medium, small) {
    this.capacity = [big, medium, small];
    this.usage = [0, 0, 0];
};

ParkingSystem.prototype.addCar = function(carType) {
    if (this.usage[carType - 1] < this.capacity[carType - 1]) {
        this.usage[carType - 1] = this.usage[carType - 1] + 1;
        return true;
    } else {
        return false;
    }
};

File: 2022/10/26/java/src/main/java/com/jarombek/ParkingSystem.java
/*
 * https://leetcode.com/problems/design-parking-system/
 * @author Andrew Jarombek
 * @since 10/26/2022
 */

package com.jarombek;

class ParkingSystem {
    private int[] spots;

    public ParkingSystem(int big, int medium, int small) {
        this.spots = new int[] {big, medium, small};
    }

    public boolean addCar(int carType) {
        int val = this.spots[carType - 1];
        this.spots[carType - 1] -= 1;
        return val > 0;
    }
}
